src/main/java/com/isa/morswiny/users/User.java

```diff
@@ -41,18 +44,19 @@ public boolean equals(Object o) {
         if (this == o) return true;
         if (!(o instanceof User)) return false;
         User user = (User) o;
-        return Objects.equals(getID(), user.getID()) &&
-                Objects.equals(getName(), user.getName()) &&
-                Objects.equals(getSurname(), user.getSurname()) &&
-                Objects.equals(getLogin(), user.getLogin()) &&
-                Objects.equals(getEmail(), user.getEmail()) &&
-                Objects.equals(getPassword(), user.getPassword()) &&
-                Objects.equals(getBirthday(), user.getBirthday());
+        return ID.equals(user.ID) &&
+                name.equals(user.name) &&
+                surname.equals(user.surname) &&
+                login.equals(user.login) &&
+                email.equals(user.email) &&
+                password.equals(user.password) &&
+                userTypes == user.userTypes &&
+                birthday.equals(user.birthday);
```

Comment:
Zastanawiam sie czy password powinno byc czescia equals. Bo tak naprawde to jeden z nielicznych parametrow, ktory moze/bedzie sie zmienial dla tego samego czlowieka. Nie chcialbym bysmy doszli do ekstremum gdzie np. tworzymy nowe konto dla tej samej osoby (wszystkie parametry takie samo ale haslo inne). Co myslicie?

Reply:
moim zdaniem juz sam login i email to jprawie identyfikacja osoby w necie :-)